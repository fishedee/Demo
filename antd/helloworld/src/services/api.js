import { stringify } from 'qs';
import request from '../utils/request';

let counter = 0;
function sleep(timeout){
  return new Promise(function(next){
      setTimeout(next,timeout);
  });
}

export async function incCounter(count){
    await sleep(100);
    counter += count;
    return counter;
}

export async function getCounter(){
    await sleep(100);
    return counter; 
}
export async function queryProjectNotice() {
  return request('/api/project/notice');
}

export async function queryActivities() {
  return request('/api/activities');
}

export async function queryRule(params) {
  return request(`/api/rule?${stringify(params)}`);
}

export async function removeRule(params) {
  return request('/api/rule', {
    method: 'POST',
    body: {
      ...params,
      method: 'delete',
    },
  });
}

export async function addRule(params) {
  return request('/api/rule', {
    method: 'POST',
    body: {
      ...params,
      method: 'post',
    },
  });
}

export async function fakeSubmitForm(params) {
  return request('/api/forms', {
    method: 'POST',
    body: params,
  });
}

export async function fakeChartData() {
  return request('/api/fake_chart_data');
}

export async function queryTags() {
  return request('/api/tags');
}

export async function queryBasicProfile() {
  return request('/api/profile/basic');
}

export async function queryAdvancedProfile() {
  return request('/api/profile/advanced');
}

export async function queryFakeList(params) {
  return request(`/api/fake_list?${stringify(params)}`);
}

export async function fakeAccountLogin(params) {
  return request('/api/login/account', {
    method: 'POST',
    body: params,
  });
}

export async function fakeRegister(params) {
  return request('/api/register', {
    method: 'POST',
    body: params,
  });
}

export async function queryNotices() {
  return request('/api/notices');
}
