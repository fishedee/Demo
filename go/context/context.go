package main

import (
	"context"
	"database/sql"
	"github.com/fishedee/app/router"
	"net/http"
)

type Context interface {
	context.Context
	DB() *Connection
}

type contextImplement struct {
	context.Context
	connection *Connection
}

func (this *contextImplement) DB() *Connection {
	return this.connection
}

func NewContext(db *sql.DB) Context {
	ctx := context.Background()
	connection := NewConnection(db, ctx)
	return &contextImplement{
		Context:    ctx,
		connection: connection,
	}
}

func NewContextMiddleware(db *sql.DB) router.RouterMiddleware {
	return func(prev router.RouterMiddlewareContext) router.RouterMiddlewareContext {
		last, isOk := prev.Handler.(func(ctx Context, w http.ResponseWriter, r *http.Request))
		if isOk == false {
			return prev
		}
		return router.RouterMiddlewareContext{
			Data: prev.Data,
			Handler: func(w http.ResponseWriter, r *http.Request, p router.RouterParam) {
				context := NewContext(db)
				last(context, w, r)
			},
		}
	}
}
