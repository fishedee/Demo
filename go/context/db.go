package main

import (
	"context"
	"database/sql"
	"fmt"
	_ "github.com/go-sql-driver/mysql"
)

type Connection struct {
	db  *sql.DB
	ctx context.Context
}

func NewConnection(db *sql.DB, ctx context.Context) *Connection {
	return &Connection{
		db:  db,
		ctx: ctx,
	}
}

func (this *Connection) Query(query string, args ...interface{}) (*sql.Rows, error) {
	return this.db.QueryContext(this.ctx, query, args...)
}

func NewDB(user string, password string, host string, port int, database string, charset string, collation string) *sql.DB {
	dblink := fmt.Sprintf(
		"%s:%s@tcp(%s:%d)/%s?charset=%v&collation=%v&loc=Local",
		user,
		password,
		host,
		port,
		database,
		charset,
		collation,
	)
	db, err := sql.Open("mysql", dblink)
	if err != nil {
		panic(err)
	}
	return db
}
