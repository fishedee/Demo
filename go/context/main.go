package main

import (
	"encoding/json"
	"fmt"
	"github.com/fishedee/app/log"
	"github.com/fishedee/app/middleware"
	"github.com/fishedee/app/router"
	"net/http"
)

type User struct {
	UserId int
	Name   string
}

func main() {
	db := NewDB("root", "1", "localhost", 3306, "Yinghao", "utf8mb4", "utf8_general_ci")
	contextMiddleware := NewContextMiddleware(db)

	log, err := log.NewLog(log.LogConfig{})
	if err != nil {
		panic(err)
	}
	logMiddleware := middleware.NewLogMiddleware(log)

	routerFactory := router.NewRouterFactory()
	routerFactory.Use(logMiddleware)
	routerFactory.Use(contextMiddleware)
	routerFactory.GET("/", func(ctx Context, w http.ResponseWriter, r *http.Request) {
		rows, err := ctx.DB().Query("select userId,name from t_user")
		if err != nil {
			panic(err)
		}
		users := []User{}
		for rows.Next() {
			var userId int
			var name string
			rows.Scan(&userId, &name)
			users = append(users, User{
				UserId: userId,
				Name:   name,
			})
		}
		data, err := json.Marshal(users)
		if err != nil {
			panic(err)
		}
		w.Write(data)
	})

	fmt.Println("server is running...")
	server := &http.Server{
		Addr:    ":8818",
		Handler: routerFactory.Create(),
	}
	err = server.ListenAndServe()
	if err != nil {
		panic(err)
	}
}
