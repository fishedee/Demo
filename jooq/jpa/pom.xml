<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
  <modelVersion>4.0.0</modelVersion>

  <groupId>com.jooq_test.app</groupId>
  <artifactId>jpa</artifactId>
  <version>1.0-SNAPSHOT</version>
  <packaging>jar</packaging>

  <name>jpa</name>
  <url>http://maven.apache.org</url>

  <parent>
    <groupId>org.springframework.boot</groupId>
    <artifactId>spring-boot-starter-parent</artifactId>
    <version>2.5.0</version>
    <relativePath/>
  </parent>

  <properties>
    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
  </properties>

  <dependencies>

    <dependency>
      <groupId>org.springframework.boot</groupId>
      <artifactId>spring-boot-starter-web</artifactId>
    </dependency>
    <dependency>
      <groupId>org.springframework.boot</groupId>
      <artifactId>spring-boot-starter-data-jpa</artifactId>
    </dependency>
    <dependency>
      <groupId>mysql</groupId>
      <artifactId>mysql-connector-java</artifactId>
      <version>8.0.18</version>
    </dependency>
    <dependency>
      <groupId>junit</groupId>
      <artifactId>junit</artifactId>
      <version>3.8.1</version>
      <scope>test</scope>
    </dependency>
    <dependency>
      <groupId>org.springframework.boot</groupId>
      <artifactId>spring-boot-starter-jooq</artifactId>
    </dependency>

    <dependency>
      <groupId>org.jooq</groupId>
      <artifactId>jooq-codegen</artifactId>
      <scope>provided</scope>
    </dependency>

    <dependency>
      <groupId>org.projectlombok</groupId>
      <artifactId>lombok</artifactId>
      <version>1.18.20</version>
      <scope>provided</scope>
    </dependency>
  </dependencies>


  <build>
    <plugins>
      <plugin>
        <groupId>org.springframework.boot</groupId>
        <artifactId>spring-boot-maven-plugin</artifactId>
        <configuration>
          <fork>true</fork>
        </configuration>
      </plugin>
      <plugin>
        <groupId>org.jooq</groupId>
        <artifactId>jooq-codegen-maven</artifactId>
        <version>3.14.9</version>
        <configuration>
          <jdbc>
            <driver>com.mysql.cj.jdbc.Driver</driver>
            <url>jdbc:mysql://127.0.0.1:3306/library?serverTimezone=GMT%2B8</url>
            <user>root</user>
            <password>1</password>
          </jdbc>
          <generator>
            <database>
              <inputSchema>library</inputSchema>
            </database>
            <target>
              <packageName>com.jooq_test.app.codegen</packageName>
              <directory>src/main/java</directory>
            </target>
          </generator>
        </configuration>
      </plugin>
    </plugins>
  </build>
</project>
