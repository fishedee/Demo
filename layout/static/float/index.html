<!doctype html>
<html>
	<head>
		<meta charset="utf-8"/>
		<title>float</title>
		<style>
			*{
				margin: 0px;
				padding: 0px;
			}
			.box{
				margin-top:30px;
				padding-bottom: 30px;
				border-bottom: 1px solid black;
			}
			.float1{
				overflow: hidden;
			}
			.float1 .item{
				width:50px;
				height: 50px;
				margin:0px 10px;
				background: yellow;
				float:left;
			}
			.inlineBox1{

			}
			.inlineBox1 .item{
				width:50px;
				height: 50px;
				margin:0px 10px;
				background: yellow;
				display: inline-block;
			}
			.float2{
				overflow: hidden;
			}
			.float2 .item1{
				float:left;
				width:50px;
				height: 50px;
				background: yellow;
				margin:0px 10px;
			}
			.float2 .item2{
				margin:0 auto;
				height: 50px;
				background: yellow;
				overflow: hidden;
			}
			.float2 .item3{
				float:right;
				width:50px;
				height: 50px;
				background: yellow;
				margin:0px 10px;
			}
			.float3{
				overflow: hidden;
				width: 220px;
				border: 1px solid blue;
			}
			.float3 .item{
				float:left;
				width:50px;
				height: 50px;
				margin:0px 10px;
				background: yellow;
			}
			.float3 .itemHigh{
				float:left;
				width:50px;
				height: 80px;
				margin:0px 10px;
				background: yellow;
			}
			.inlineBox3{
				width: 220px;
				border: 1px solid blue;
			}
			.inlineBox3 .item{
				display: inline-block;
				width:50px;
				height: 50px;
				margin:0px 10px;
				background: yellow;
			}
			.inlineBox3 .itemHigh{
				display: inline-block;
				width:50px;
				height: 70px;
				margin:0px 10px;
				background: yellow;
			}
			.float4{
				overflow: hidden;
				border: 1px solid blue;
			}
			.float4 .item{
				float:left;
				width:50px;
				height: 50px;
				background: yellow;
			}
			.float4 p{
				width: 400px;
			}
			.inlineBox4{
				border: 1px solid blue;
			}
			.inlineBox4 .item{
				display: inline-block;
				width:50px;
				height: 50px;
				background: yellow;
				vertical-align: top;
			}
			.inlineBox4 p{
				display: inline-block;
				width: 400px;
			}
		</style>
	</head>
	<body>
		<div class="box">
			<h1>float on one line</h1>
			<div class="float1">
				<div class="item">a</div>
				<div class="item">b</div>
				<div class="item">c</div>
			</div>
			<h1>inline-box on one line</h1>
			<div class="inlineBox1">
				<div class="item">a</div><div class="item">b</div><div class="item">c</div>
			</div>
		</div>
		<div class="box">
			<h1>float right</h1>
			<div class="float2">
				<div class="item1">a</div>
				<div class="item3">c</div>
				<div class="item2">b</div>
			</div>
		</div>
		<div class="box">
			<h1>float on more higher</h1>
			<div class="float3">
				<div class="item">a</div>
				<div class="itemHigh">b</div>
				<div class="item">c</div>
				<div class="item">d</div>
			</div>
			<h1>inline-box on more higher</h1>
			<div class="inlineBox3">
				<div class="item">a</div><div class="itemHigh">b</div><div class="item">c</div><div class="item">d</div>
			</div>
		</div>
		<div class="box">
			<h1>float on wrap</h1>
			<div class="float4">
				<div class="item">a</div>
				<p>你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊</p>
			</div>
			<h1>inline-box on wrap</h1>
			<div class="inlineBox4">
				<div class="item">a</div>
				<p>你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊你好啊</p>
			</div>
		</div>
	</body>
</html>