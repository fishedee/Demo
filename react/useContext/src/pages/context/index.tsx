import { createContext, useContext, useState, memo } from 'react';

const ModeContext = createContext({ mode: 'fish' });

let GrandSon = memo(function () {
    console.log('grand son render');
    let data = useContext(ModeContext);
    return (
        <div>
            <div>我是孙组件</div>
            <div>mode为:{data.mode}</div>
        </div>
    );
});
let Son = memo(function () {
    console.log('son render');
    return (
        <div>
            <div>我是子组件</div>
            <GrandSon />
        </div>
    );
});
export default function () {
    console.log('top render');
    let [mode, setMode] = useState('fish');
    return (
        <div>
            <h3>当前mode为:{mode}</h3>
            <button
                onClick={() => {
                    setMode((mode) => {
                        if (mode == 'fish') {
                            return 'cat';
                        } else {
                            return 'fish';
                        }
                    });
                }}
            >
                切换mode
            </button>
            <ModeContext.Provider value={{ mode: mode }}>
                <Son />
            </ModeContext.Provider>
        </div>
    );
}
