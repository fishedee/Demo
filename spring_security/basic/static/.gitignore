node_modules
.DS_Store
dist/main-*.js
dist/vendors-*.js
dist/webpack-*.js
