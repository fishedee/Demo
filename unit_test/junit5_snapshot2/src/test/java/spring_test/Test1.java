package spring_test;

import au.com.origin.snapshots.junit5.SnapshotExtension;
import com.fasterxml.jackson.databind.ObjectMapper;
import org.junit.jupiter.api.*;

import static org.junit.jupiter.api.Assertions.*;

import org.junit.jupiter.api.extension.ExtendWith;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.autoconfigure.jackson.JacksonAutoConfiguration;
import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.context.annotation.Import;
import au.com.origin.snapshots.Expect;

/**
 * Created by fish on 2021/5/30.
 */
@SpringBootTest
@Import({JacksonAutoConfiguration.class})
@ExtendWith({SnapshotExtension.class})
public class Test1 {
    private Expect expect;


    @Autowired
    private ObjectMapper objectMapper;

    public String toString(Object data)throws Exception{
        return objectMapper.writeValueAsString(data);
    }

    //字符串测试，单用例下的多测试
    @Test
    public void test1() throws Exception{

        OrderDO orderDO = new OrderDO();

        orderDO.setSize(1);
        orderDO.setName("fish");
        expect.scenario("1").toMatchSnapshot(toString(orderDO));


        OrderDO orderDO2 = new OrderDO();

        orderDO2.setSize(2);
        orderDO2.setName("dog");

        expect.scenario("2").toMatchSnapshot(toString(orderDO2));

    }

    @Test
    public void test2() throws Exception{

        OrderDO orderDO = new OrderDO();

        orderDO.setSize(1);
        orderDO.setName("fish");
        expect.scenario("1").serializer("json").toMatchSnapshot(toString(orderDO));


        OrderDO orderDO2 = new OrderDO();

        orderDO2.setSize(2);
        orderDO2.setName("dog");

        expect.scenario("2").serializer("json").toMatchSnapshot(toString(orderDO2));

    }

}
