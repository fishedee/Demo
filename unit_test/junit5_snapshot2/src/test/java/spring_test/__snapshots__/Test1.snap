spring_test.Test1.test1[1]=[
{"id":null,"name":"fish","email":null,"size":1,"total":null}
]


spring_test.Test1.test1[2]=[
{"id":null,"name":"dog","email":null,"size":2,"total":null}
]


spring_test.Test1.test2[1]=[
  "{\"id\":null,\"name\":\"fish\",\"email\":null,\"size\":1,\"total\":null}"
]


spring_test.Test1.test2[2]=[
  "{\"id\":null,\"name\":\"dog2\",\"email\":null,\"size\":2,\"total\":null}"
]